### PR TITLE
Add Vicious40 to QMK

### PR DESCRIPTION
**Pull request by awkannan**
## Description

Adding the Vicious40 PCB to QMK

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/cannonkeys/vicious40/rules.mk [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+NKRO_ENABLE = yes	    # USB Nkey Rollover
+CUSTOM_MATRIX = no # Custom matrix file
+BACKLIGHT_ENABLE = yes
+RGBLIGHT_ENABLE = no
+
+# Enter lower-power sleep mode when on the ChibiOS idle thread
+OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE
+
+# Wildcard to allow APM32 MCU
+DFU_SUFFIX_ARGS = -p FFFF -v FFFF
```
Comment (fauxpark) [inline keyboards/cannonkeys/vicious40/rules.mk:22]:
```suggestion
# Wildcard to allow APM32 MCU
DFU_SUFFIX_ARGS = -p FFFF -v FFFF

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = yes      # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output

# Enter lower-power sleep mode when on the ChibiOS idle thread
OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE
```

File: keyboards/cannonkeys/vicious40/readme.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# CannonKeys Vicious40
+
+Vicious40 Keyboard
+
+Keyboard Maintainer: [Andrew Kannan](https://github.com/awkannan)  
+Hardware Supported: STM32F072CBT6 or APM32F072CBT6
```
Comment (fauxpark) [inline keyboards/cannonkeys/vicious40/readme.md:6]:
```suggestion
* Keyboard Maintainer: [Andrew Kannan](https://github.com/awkannan)
* Hardware Supported: STM32F072CBT6 or APM32F072CBT6
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/cannonkeys/vicious40/keymaps/via/rules.mk [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# rules.mk overrides to enable VIA
+
+VIA_ENABLE = yes
+
```
Comment (fauxpark) [inline keyboards/cannonkeys/vicious40/keymaps/via/rules.mk:4]:
```suggestion
VIA_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by awkannan** [comment]
Should be all set for another look!

File: keyboards/cannonkeys/vicious40/rules.mk [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# MCU name
+MCU = STM32F072
+
+# Wildcard to allow APM32 MCU
```
Comment (fauxpark) [inline keyboards/cannonkeys/vicious40/rules.mk:4]:
```suggestion
# Bootloader selection
BOOTLOADER = stm32-dfu

# Wildcard to allow APM32 MCU
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/cannonkeys/vicious40/info.json [hunk 5]
```diff
@@ -0,0 +1,12 @@
+{
+    "keyboard_name": "Vicious40",
+    "url": "https://cannonkeys.com",
+    "maintainer": "awkannan",
+    "width": 12.75,
+    "height": 4,
+    "layouts": {
+        "LAYOUT": {
```
Comment (fauxpark) [inline keyboards/cannonkeys/vicious40/info.json:8]:
```suggestion
        "LAYOUT_default": {
```

**Review by fauxpark — CHANGES_REQUESTED** [review]
Whoops, one more thing then looks good to go.

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**